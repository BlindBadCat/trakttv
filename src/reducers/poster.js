import C from '../constants';

export const initialState = {
  isFetching: false,
  urls: [],
};

const poster = (state = initialState, { type, payload }) => {
  /*switch (type) {
    case C.GET_POSTER_URL_REQUEST:
      return { ...state, isFetching: true };
    case C.GET_POSTER_URL_ADN_ID:
      return { ...state, urls: [...state.urls, payload] };
    case C.RESET_FOR_NEW_FETCH:
      return { ...initialState };
    default: return state;
  }*/
  return state;
};


export default poster;
