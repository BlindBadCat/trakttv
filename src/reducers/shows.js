import C from '../constants';

export const initialState = {
  isFetching: false,
  shows: [],
  sort: 'shows/watched/all',
  query: '',
  genre: '',
  pageCount: 5,
  limit: 10,
  currentPage: 1,
  itemCount: 10,
};

const shows = (state = initialState, { type, payload }) => {
  switch (type) {
    case C.FETCH_SHOWS_REQUEST:
      return { ...state, isFetching: true };
    case C.FETCH_SHOWS_SUCCESS:
      return {
        ...state,
        shows: [...payload.shows],
        ...payload.pagination,
        isFetching: false,
      };
    case C.FETCH_SHOWS_ERROR:
      return { ...state };
    case C.CHANGE_URL_PARAMS:
      return { ...state, ...payload };
    default: return state;
  }
};

export default shows;
