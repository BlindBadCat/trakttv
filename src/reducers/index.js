import { combineReducers } from 'redux';
import { routerReducer } from 'react-router-redux';
import shows from './shows';

const rootReducer = combineReducers({
  routerReducer,
  shows,
});

export default rootReducer;
