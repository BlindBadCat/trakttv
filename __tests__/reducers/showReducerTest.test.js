/* eslint-disable import/no-duplicates */
import deepFreeze from 'deep-freeze';
import shows from '../../src/reducers/shows';
import C from '../../src/constants/index';
import { initialState } from '../../src/reducers/shows';

describe('Tests for shows reducer', () => {

  it('should return the initial state', () => {
    const action = {};
    deepFreeze(initialState);
    deepFreeze(action);
    expect(shows(undefined, action)).toEqual(initialState);
  });

  it('Should handle wrong action', () => {
    const action = { type: C.GET_POSTER_URL_ADN_ID };
    const state = { ...initialState };
    const result = { ...initialState };
    expect(shows(state, action)).toEqual(result);
  });

  it('should handle GET_SHOWS_REQUEST', () => {
    const action = { type: C.GET_SHOWS_REQUEST };
    const result = { ...initialState, isFetching: true };
    deepFreeze(initialState);
    deepFreeze(action);
    expect(shows(initialState, action)).toEqual(result);
  });

  it('should hande GET_SHOWS_ERROR', () => {
    const action = { type: C.GET_SHOWS_ERROR, payload: 'some error' };
    const result = { ...initialState };
    deepFreeze(initialState);
    deepFreeze(action);
    expect(shows(initialState, action)).toEqual(result);
  });

  it('should hande RESET_SHOWS_FOR_PAGINATION', () => {
    const state = {
      isFetching: true,
      shows: [{}, {}, {}, {}],
      urlParams: {
        searchUrl: 'shows/watched/all',
        query: '',
        genre: '',
        pagination: {
          pageCount: 15,
          limit: 10,
          currentPage: 10,
          itemCount: 150,
        },
      },
    };

    const action = { type: C.RESET_SHOWS_FOR_PAGINATION };

    const result = {
      isFetching: false,
      shows: [],
      urlParams: {
        searchUrl: 'shows/watched/all',
        query: '',
        genre: '',
        pagination: {
          pageCount: 15,
          limit: 10,
          currentPage: 10,
          itemCount: 150,
        },
      },
    };

    deepFreeze(state);
    deepFreeze(action);
    expect(shows(state, action)).toEqual(result);
  });

  it('should handle ADD_SHOWS if they are present', () => {
    const action = {
      type: C.ADD_SHOWS,
      payload: {
        shows: [
          { someValue: 1, show: { title: 'show1' } },
          { someValue: 2, show: { title: 'show2' } },
          { someValue: 3, show: { title: 'show3' } },
          { someValue: 4, show: { title: 'show4' } },
        ],
        urlParams: {
          searchUrl: 'shows/watched/all',
          query: '',
          genre: '',
          pagination: {
            pageCount: 4000,
            limit: 10,
            currentPage: 1,
            itemCount: 40000,
          },
        },
      },
    };

    const result = {
      isFetching: true,
      shows: [
        { title: 'show1' },
        { title: 'show2' },
        { title: 'show3' },
        { title: 'show4' },
      ],
      urlParams: {
        searchUrl: 'shows/watched/all',
        query: '',
        genre: '',
        pagination: {
          pageCount: 4000,
          limit: 10,
          currentPage: 1,
          itemCount: 40000,
        },
      },
    };
    const state = { ...initialState, isFetching: true };
    deepFreeze(action);
    deepFreeze(state);
    expect(shows(state, action)).toEqual(result);
  });

});
