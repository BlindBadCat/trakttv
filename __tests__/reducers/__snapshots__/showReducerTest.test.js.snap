// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for shows reducer should handle CHANGE_URL_PARAMS 1`] = `
Object {
  "currentPage": 5,
  "genre": "",
  "isFetching": false,
  "itemCount": 10,
  "limit": 10,
  "pageCount": 5,
  "query": "",
  "shows": Array [],
  "sort": "shows/watched/all",
}
`;

exports[`Tests for shows reducer should handle FETCH_SHOWS_ERROR 1`] = `
Object {
  "currentPage": 1,
  "genre": "",
  "isFetching": false,
  "itemCount": 10,
  "limit": 10,
  "pageCount": 5,
  "query": "",
  "shows": Array [],
  "sort": "shows/watched/all",
}
`;

exports[`Tests for shows reducer should handle FETCH_SHOWS_REQUEST 1`] = `
Object {
  "currentPage": 1,
  "genre": "",
  "isFetching": true,
  "itemCount": 10,
  "limit": 10,
  "pageCount": 5,
  "query": "",
  "shows": Array [],
  "sort": "shows/watched/all",
}
`;

exports[`Tests for shows reducer should handle FETCH_SHOWS_SUCCESS 1`] = `
Object {
  "currentPage": 1,
  "genre": "",
  "isFetching": false,
  "itemCount": 40000,
  "limit": 10,
  "pageCount": 4000,
  "query": "",
  "shows": Array [
    Object {
      "show": Object {
        "title": "show1",
      },
      "someValue": 1,
    },
    Object {
      "show": Object {
        "title": "show2",
      },
      "someValue": 2,
    },
    Object {
      "show": Object {
        "title": "show3",
      },
      "someValue": 3,
    },
    Object {
      "show": Object {
        "title": "show4",
      },
      "someValue": 4,
    },
  ],
  "sort": "shows/watched/all",
}
`;

exports[`Tests for shows reducer should handle wrong actions 1`] = `
Object {
  "currentPage": 1,
  "genre": "",
  "isFetching": false,
  "itemCount": 10,
  "limit": 10,
  "pageCount": 5,
  "query": "",
  "shows": Array [],
  "sort": "shows/watched/all",
}
`;

exports[`Tests for shows reducer should return the initial state 1`] = `
Object {
  "currentPage": 1,
  "genre": "",
  "isFetching": false,
  "itemCount": 10,
  "limit": 10,
  "pageCount": 5,
  "query": "",
  "shows": Array [],
  "sort": "shows/watched/all",
}
`;
